app/services/foreman/renderer/scope/macros/snippet_rendering.rb

```diff
@@ -46,9 +62,15 @@ def snippet_if_exists(name, options = {})
             example "snippet('ntp_server', variables: { :enable_ntp => true) } # => '...'"
           end
           def snippet(name, options = {}, variables: {})
-            template = source.find_snippet(name)
+            if name.is_a?(Array)
+              tpl_name = name.find { |tpl| snippet_exists(tpl) } || ''
+            else
+              tpl_name = name
+            end
+
+            template = source.find_snippet(tpl_name)
```

Comment:
This feels inefficient. You're going to call `find_snippet` on every entry and then another time.

I wonder if `find_snippet` itself should be enhanced to deal with a list of names.

Reply:
I'm open to the idea of reworking this in that direction but was trying to keep changes to the existing api/function calls as minimally intrusive as possible.